x/datapool/types/pool.go

```diff
@@ -33,45 +29,8 @@ func NewPool(poolID uint64, curator sdk.AccAddress, poolParams PoolParams) *Pool
 }
 
 func NewPoolAddress(poolID uint64) sdk.AccAddress {
-	key := append([]byte("pool"), sdk.Uint64ToBigEndian(poolID)...)
-	return address.Module(ModuleName, key)
-}
-
-func AccPoolAddressFromBech32(address string) (sdk.AccAddress, error) {
-	if len(strings.TrimSpace(address)) == 0 {
-		return sdk.AccAddress{}, fmt.Errorf("empty address string is not allowed")
-	}
-
-	bech32PrefixAccAddr := sdk.GetConfig().GetBech32AccountAddrPrefix()
-
-	bz, err := sdk.GetFromBech32(address, bech32PrefixAccAddr)
-	if err != nil {
-		return nil, err
-	}
-
-	err = verifyPoolAddressFormat(bz)
-	if err != nil {
-		return nil, err
-	}
-
-	return sdk.AccAddress(bz), nil
-}
-
-func verifyPoolAddressFormat(bz []byte) error {
-	if len(bz) == 0 {
-		return sdkerrors.Wrap(sdkerrors.ErrUnknownAddress, "address cannot be empty")
-	}
-
-	if len(bz) > address.MaxAddrLen {
-		return sdkerrors.Wrapf(sdkerrors.ErrUnknownAddress, "address max length is %d, got %d", address.MaxAddrLen, len(bz))
-	}
-
-	verifier := sdk.GetConfig().GetAddressVerifier()
-	if verifier != nil {
-		return verifier(bz)
-	}
-
-	return nil
+	poolKey := "pool" + strconv.FormatUint(poolID, 10)
+	return authtypes.NewModuleAddress(poolKey)
```

Comment:
If the length of the address in the pool and the length of the general address are the same, is there any problem?
`address.Module(ModuleName, key)` length is 20.
But `authtypes.NewModuleAddress(poolKey)` length is 32.

Reply:
Thank you for your double-checking.
IIRC, this kind of module address is also used in `x/datadeal` and there was no specific reason to use 32-length address. 

Rather, 20-length address should be used because wasm keeper checks the length of address is 20. Since module & pool account address is used as a minter in wasm keeper, it makes sense to me.

Reply:
If higher version of wasm keeper can take longer address, we can also use it :)

Reply:
Ah, wasm checks it.
Then I'd like to go this way.